parquet-column/src/main/java/org/apache/parquet/column/statistics/Statistics.java

```diff
@@ -150,14 +191,17 @@ public int hashCode() {
   public void mergeStatistics(Statistics stats) {
     if (stats.isEmpty()) return;
 
-    if (this.getClass() == stats.getClass()) {
+    // Merge stats only if they have the same type and comparator (the sorting order
+    // is the same)
+    if (this.getClass() == stats.getClass() && type.getPrimitiveTypeName() == stats.type.getPrimitiveTypeName()
+        && type.getOriginalType() == stats.type.getOriginalType()) {
```

Comment:
Why check the primitive type and logical type instead of delegating to the type's equals method? This implementation doesn't catch attempting to merge `fixed[4]` and `fixed[2]`, for example. I think this should keep the type and delegate.

Reply:
Currently, we have many places where a `Statistics` class is created by using its public constructors (which are deprecated already, but still supported). These constructors does not have any `PrimitiveType` objects therefore we use fake ones. It means that we are not able to use the `PrimitiveType`'s `equals` method because it checks e.g. the name of the column.

Reply:
Are the statistics created by these constructors used in the write path? I think it is fine to require that statistics are created with the same type, whether that type is defaulted or passed in.